webpack/index.d.ts

```diff
@@ -654,7 +654,14 @@ declare namespace webpack {
      */
 
     class BannerPlugin extends Plugin {
-        constructor(banner: any, options: any);
+        constructor(options: {
+          banner: string,
+          raw?: boolean,
+          entryOnly?: boolean,
+          test?: string | RegExp | Array<string | RegExp>,
+          include?: string | RegExp | Array<string | RegExp>,
+          exclude?: string | RegExp | Array<string | RegExp>
```

Comment:
i'd
- order the keys alphabetically
- optionally create type aliases for `string | RegExp` and `string | RegExp | Array<string | RegExp>`
- optionally add a Options interface under a namespace like [this](https://github.com/tommytroylin/DefinitelyTyped/blob/592b7f26f90dc6013c43e9be2b0966fc11dbc42a/webpack/index.d.ts#L793-L808). This way, users can `import { BannerPlugin } ...` and use `BannerPlugin.Options`.